org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListbox.java

```diff
@@ -1255,5 +1277,99 @@ public void setAllowIDColumnForReadWrite(boolean allowIDColumnForReadWrite)
 	{
 		this.allowIDColumnForReadWrite = allowIDColumnForReadWrite;
 	}
+	
+	   /**
+     * Retrieve the width of the column
+     * 
+     * @param columnName the name of the column to find the width of.
+     * @param columnList list of the columns and their width
+     * @return the width of the column
+     */
+	private String getColumnWidth(String columnName, List<String>columnList) 
+	{
+		String width ="null";
+		String[] w;
+		
+		for (String column : columnList)
+		{
+			w = column.trim().split("=");
+			if (w[0].equals(columnName))
+			{
+				return w[1];
+			}
+		}
+		return width;
+	}
+	
+    /**
+     * Render the header of the WListbox with the columns width of the saved column width
+     * 
+     */
+	public void renderCustomHeaderWidth () 
+	{
+		if (wListBoxName != null && getListHead() != null) 
+		{
+			MWlistboxCustomization WListBoxCustomization =    MWlistboxCustomization.get(Env.getCtx(), Env.getAD_User_ID(Env.getCtx()), wListBoxName, null);
+			Boolean isHasCustomizeData =  WListBoxCustomization != null &&
+					  WListBoxCustomization.getAD_Wlistbox_Customization_ID() > 0 &&
+					  WListBoxCustomization.getCustom() != null &&
+					  WListBoxCustomization.getCustom().trim().length() > 0; 			
+			if (isHasCustomizeData) 
+			{
+				String[] custom = WListBoxCustomization.getCustom().trim().split(",");		
+				List<String>columnList = Arrays.asList(custom);
+				ColumnInfo[] Columns = this.getLayout();
+				Listhead listHead =  getListhead();
+				if (listHead != null && columnList.size() > 0) 
+				{
+					List<?> headers = listHead.getChildren();
+					int i = 0;
+					for(Object obj : headers)
+					{
+						Listheader header = (Listheader) obj;
+						String columnWidth = this.getColumnWidth(Columns[i].getColumnName(), columnList);
+						if (!("null".equals(columnWidth)))	
+							ZKUpdateUtil.setWidth(header, columnWidth);
+						i++; 
+					}
+				}
+			}
+		}
+	}
+
+	/**
+     * Save the width of all the columns in the WLsitbox  
+     */
+	public void saveColumnWidth() 
+	{
+		StringBuilder custom = new StringBuilder(); 
+		String width = "";
+		String colName = "";
+		Listhead listHead = getListHead();
+		ColumnInfo[] layout = getLayout();
+		if (listHead != null && layout != null && !Util.isEmpty(getwListBoxName())) 
+		{
+			List<?> headers = listHead.getChildren();
+			int i = 0;
+			for(Object obj : headers)
+			{
+				Listheader header = (Listheader) obj;
+				width = header.getWidth();
+				colName = layout[i].getColumnName();
```

Comment:
needs to check whether columnName is null/empty here.

Reply:
Done